x-pack/plugins/rule_registry/common/search_strategy/index.ts

```diff
@@ -27,51 +25,11 @@ export interface RuleRegistrySearchRequestPagination {
   pageSize: number;
 }
 
-type Prev = [
-  never,
-  0,
-  1,
-  2,
-  3,
-  4,
-  5,
-  6,
-  7,
-  8,
-  9,
-  10,
-  11,
-  12,
-  13,
-  14,
-  15,
-  16,
-  17,
-  18,
-  19,
-  20,
-  ...Array<0>
-];
-
-type Join<K, P> = K extends string | number
-  ? P extends string | number
-    ? `${K}${'' extends P ? '' : '.'}${P}`
-    : never
-  : string;
-
-type DotNestedKeys<T, D extends number = 10> = [D] extends [never]
-  ? never
-  : T extends object
-  ? { [K in keyof T]-?: Join<K, DotNestedKeys<T[K], Prev[D]>> }[keyof T]
-  : never;
-
-export type EcsFields = DotNestedKeys<Omit<Ecs, 'ecs'>>;
-
 export interface BasicFields {
   _id: string;
   _index: string;
 }
-export type EcsFieldsResponse = {
-  [Property in EcsFields]: string[];
-} & BasicFields;
+export type EcsFieldsResponse = BasicFields & {
+  [x: string]: unknown[];
+};
```

Comment:
It looks like this change is causing the type check to fail. Essentially, `_id` and `_index` coming from `BasicFields` do not match the index signature `{ [x: string]: unknown[] }`.

However, leaving the `EcsFields` type does not help either because `kibana.alert.rule.name` does not exist either in the base `Ecs`, so it'll fail anyway.

Any idea of how to solve this issue?